pkg/scannerv4/mappers/mappers.go

```diff
@@ -658,45 +705,69 @@ type severityValues struct {
 	v2Score  float32
 	v3Vector string
 	v3Score  float32
+	source   v4.VulnerabilityReport_Vulnerability_CVSS_Source
+	url      string
 }
 
-// severityAndScores returns the severity and scores information out of a
-// ClairCore vulnerability. The returned information is dependent on the
-// underlying updater.
-func severityAndScores(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) (severityValues, error) {
+// severityAndCvssMetrics processes the severity and CVSS scores
+// for a given vulnerability based on the updater type (RHEL, NVD, OSV, etc.).
+// The returned data includes CVSS metrics, ACS preferred severity values, and potential errors
+// encountered during the calculation process for each source.
+func severityAndCvssMetrics(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) ([]*v4.VulnerabilityReport_Vulnerability_CVSS, severityValues, error) {
 	ctx = zlog.ContextWithValues(ctx,
-		"component", "mappers/severityAndScores",
+		"component", "mappers/severityAndCvssMetrics",
 		"updater", vuln.Updater,
 		"vuln_id", vuln.ID,
 		"vuln_name", vuln.Name,
 	)
-
+	var cvssMetrics []*v4.VulnerabilityReport_Vulnerability_CVSS
+
+	var sevs []severityValues
+	rhelSev, rhelErr := rhelSeverityAndScores(vuln)
+	if rhelErr == nil {
+		sevs = append(sevs, rhelSev)
+	}
+	nvdSev, nvdErr := nvdSeverityAndScores(vuln, nvdVuln)
+	if nvdErr == nil {
+		sevs = append(sevs, nvdSev)
+	}
+	undecidedSev, undecidedErr := cvssVector(vuln.Severity)
+	if undecidedErr == nil {
+		if strings.HasPrefix(vuln.Updater, osvUpdaterPrefix) {
+			undecidedSev.source = v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_OSV
+			undecidedSev.url = vuln.Links
+			sevs = append(sevs, undecidedSev)
+		} else if strings.EqualFold(vuln.Updater, constants.ManualUpdaterName) {
+			undecidedSev.source = v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_UNKNOWN
+			undecidedSev.url = vuln.Links
+			sevs = append(sevs, undecidedSev)
+		}
+	}
+	cvssMetrics = toCVSSMetrics(ctx, sevs, cvssMetrics)
 	switch {
 	case rhelUpdaterPattern.MatchString(vuln.Updater):
-		return rhelSeverityAndScores(vuln)
+		return cvssMetrics, rhelSev, rhelErr
 	case strings.HasPrefix(vuln.Updater, osvUpdaterPrefix):
-		if isOSVDBSpecificSeverity(vuln.Severity) {
+		if isOSVDBSpecificSeverity(vuln.Severity) || undecidedErr != nil {
 			zlog.Debug(ctx).
 				Str("severity", vuln.Severity).
 				Msg("severity is not a CVSS vector, falling back to NVD")
 			break
 		}
-		// The severity may be a CVSS vector, so fallthrough to check.
-		fallthrough
+		return cvssMetrics, undecidedSev, nil
```

Comment:
By adding `undecidedErr` to this condition (and removing `fallthrough`) we no longer have a log for the `undecidedErr` - why remove the fallthrough? - consider adding it back or logging the error